pennylane/grouping/optimize_measurements.py

```diff
@@ -0,0 +1,95 @@
+from pennylane.grouping.transformations import obtain_qwc_post_rotations_and_diagonalized_groupings
+
+
+def optimize_measurements(observables, coefficients=None, grouping="qwc", colouring_method="rlf"):
+    """Partitions then diagonalizes a list of Pauli words, facilitating simultaneous measurement of
+    all observables within a partition.
+
+    The input list of observables are partitioned into mutually qubit-wise commuting (QWC) or
+    mutually commuting partitions by approximately solving minimum clique cover on a graph where
+    each observable represents a vertice. The unitaries which diagonalize the partitions are then
+    found. See arXiv:1907.03358 and arXiv:1907.09386 for technical details of the QWC and fully
+    commuting measurement partitioning approaches respectively.
+
+    **Example usage:**
+
+    >>> observables = [qml.PauliY(0), qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(1)]
+    >>> coefficients = [1.43, 4.21, 0.97]
+    >>> post_rotations,diagonalized_groupings,grouped_coeffs = optimize_measurements(
+                                                                                     observables,
+                                                                                     coefficients,
+                                                                                     'qwc',
+                                                                                     'rlf'
+                                                                                     )
+    >>> print(post_rotations)
+    [[RY(-1.5707963267948966, wires=[0]), RY(-1.5707963267948966, wires=[1])],
+     [RX(1.5707963267948966, wires=[0])]]
+    >>> print(diagonalized_groupings)
+    [[Tensor(PauliZ(wires=[0]), PauliZ(wires=[1]))],
+     [Tensor(PauliZ(wires=[0])), Tensor(PauliZ(wires=[1]))]]
+    >>> print(grouped_coeffs)
+    [[4.21], [1.43, 0.97]]
+
+    Args:
+        observables (list[Observable]): a list of Pauli words (Pauli operation instances and Tensor
+            instances thereof).
+
+    Keyword args:
+        coefficients (list[scalar]): a list of scalar coefficients.
+        grouping (str): the binary symmetric relation to use for operator partitioning.
+        colouring_method (str): the graph colouring heuristic to use in obtaining the operator
+            partitions.
+
+    Returns:
+        post_rotations (list[Template]): a list of the post-rotation qml.Templates instances, one
+            for each partition.
+        diagonalized_groupings (list[list[Observable]]): a list of the obtained groupings. Each
+            grouping is itself a list of Pauli words diagonal in the measurement basis.
+        grouped_coeffs (list[list[scalar]]): a list of coefficient groupings. Each
+            coefficient grouping is itself a list of the partitions corresponding coefficients.
+            (Only output if coefficients are specified.)
+
+    """
+
+    if coefficients is None:
```

Comment:
group_observables already allows coefficients = None, so I don't think you need the if-else here.

Reply:
Currently the if-else is used to catch the varying output structure of `group_observables`. If `coefficients` is `None`, then `group_observables` only outputs the list of partitioned Pauli words. Otherwise, `group_observables` outputs the list of partitioned Pauli words, along with the list of partitioned coefficients, `(partitioned_paulis, partitioned_coefficients)`. Should this be changed?

Reply:
No, you're right. Let's keep it as-is for now.